src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -98,22 +98,13 @@ internal sealed class ContainedDocument : ForegroundThreadAffinitizedObject, IVi
             _optionService = _workspace.Services.GetService<IOptionService>();
             _hostType = GetHostType();
 
-            var rdt = (IVsRunningDocumentTable)componentModel.GetService<SVsServiceProvider>().GetService(typeof(SVsRunningDocumentTable));
-
-            IVsHierarchy sharedHierarchy;
-            uint itemIdInSharedHierarchy;
-            var isSharedHierarchy = LinkedFileUtilities.TryGetSharedHierarchyAndItemId(hierarchy, itemId, out sharedHierarchy, out itemIdInSharedHierarchy);
-
-            var filePath = isSharedHierarchy
-                ? rdt.GetMonikerForHierarchyAndItemId(sharedHierarchy, itemIdInSharedHierarchy)
-                : rdt.GetMonikerForHierarchyAndItemId(hierarchy, itemId);
-
-            // we couldn't look up the document moniker in RDT for a hierarchy/item pair
-            // Since we only use this moniker as a key, we could fall back to something else, like the document name.
-            if (filePath == null)
+            string filePath;
+            if (!ErrorHandler.Succeeded(((IVsProject)hierarchy).GetMkDocument(itemId, out filePath)))
```

Comment:
Actually, TypeScript is IVsProject3.  And we do implement GetMkDocument on it:
https://github.com/Microsoft/TypeScript-VisualStudio/blob/master/LanguageService/TypeScriptVisualStudio/VirtualProjects/VsVirtualProjectsHierarchy.IVsProject3.cs

Pinging @jramsay.  Jason, do you see a problem with this with your Venus/Contained-language work?